src/protocol/attribution/input.rs

```diff
@@ -87,26 +89,59 @@ impl<F: Field, T: LinearSecretSharing<F>> MCAccumulateCreditInputRow<F, T> {
     pub fn new(
         is_trigger_report: T,
         helper_bit: T,
+        active_bit: T,
         breakdown_key: Vec<T>,
         trigger_value: T,
     ) -> Self {
         Self {
             is_trigger_report,
             helper_bit,
+            active_bit,
             breakdown_key,
             trigger_value,
             _marker: PhantomData,
         }
     }
 }
 
-pub type MCAccumulateCreditOutputRow<F, T> = MCAccumulateCreditInputRow<F, T>;
+pub type AccumulateCreditOutputRow<F, BK> = CreditCappingInputRow<F, BK>;
+pub type MCAccumulateCreditOutputRow<F, T> = MCCreditCappingInputRow<F, T>;
 
 //
 // `credit_capping` protocol
 //
-pub type CreditCappingInputRow<F, BK> = AccumulateCreditInputRow<F, BK>;
-pub type MCCreditCappingInputRow<F, T> = MCAccumulateCreditInputRow<F, T>;
+pub struct CreditCappingInputRow<F: Field, BK: GaloisField> {
```

Comment:
@martinthomson - what's your opinion about all of these various struct definitions for each step of the protocol?

Personally, I am not in love with it. I think they have the effect of making the code look more complicated than it actually is. We also find ourselves serialising back and forth quite a bit from vectors of these structs, to vectors of just one element from the struct.

I also really don't like how much code these create, for all of the `reshare` and `upgrade` and test input, and reconstruct functions.

I think I'm in favor of just eliminating 100% of these structs, and simply passing around vectors of each of the things inside of them.

e.g.

```
pub async fn accumulate_credit<F, C, T>(
    ctx: C,
//    input: &[MCAccumulateCreditInputRow<F, T>], eliminate this
    is_trigger_report_bits: &[T],
    helper_bits: &[T],
//    breakdown_keys: &[T] I don't think this is even needed for accumulate credit step
    trigger_values: &[T],
    stop_bits: &[T],
    per_user_credit_cap: u32,
    attribution_window_seconds: u32,
) -> Result<Vec<MCAccumulateCreditOutputRow<F, T>>, Error>
where
    F: Field,
    C: Context,
    T: LinearSecretSharing<F> + BasicProtocols<C, F>,
{
```

Reply:
I'm not sure that there is an easy answer here.  I definitely agree that transforming stuff is going to be costing us more than is good for us, but massive function signatures and no encapsulation has some other consequences that aren't great either.

If there are places where these inputs can be logically grouped in ways that work across multiple stages of the computation, then we might do what you have there, with some groupings that make sense.